scripts/common.sh

```diff
@@ -33,6 +33,79 @@ else
     function printLog() { echo "$(tput setaf 3)$1$(tput sgr0)"; }
 fi
 
+function fail() {
+    printError "$@"
+    return 1
```

Comment:
is this essentially the same as `exit 1`?

Reply:
Same but better :) With `exit 1` you immediately exit from the script. With `return 1` you just return with a non-zero status giving the calling script control over what happens next. If the script has `set -e` (which all of our scripts either do or should be doing), it will exit by default but you can also silence the error with `|| true` or `set +e` and do some error handling if you need to.

Basically, this makes the error work more like throwing an exception which bubbles up the stack than a hard exit.